### ISSUE TITLE
[Bug]: First time startup does not display any indicator that a setup is occuring

### ISSUE THREAD

**Issue by amanape** [All-Hands-AI/OpenHands#3551]
### Is there an existing issue for the same bug?

- [X] I have checked the troubleshooting document at https://docs.all-hands.dev/modules/usage/troubleshooting
- [X] I have checked the existing issues.

### Describe the bug

When first starting a new session, the terminal stops at `12:38:13 - openhands:INFO: docker.py:75 - Cannot pull image ghcr.io/all-hands-ai/runtime:a6d8c497e8c9ea9ad7786861232e1aff directly` for the next ~7 minutes when it is actually working in the background. This makes it look like an error occurred and may cause the user to force-quit.

### Current OpenHands version

```bash
0.8.3
```

### Installation and Configuration

```bash
Clear any previous Docker images and containers related to OpenHands. `make build && make run` and `Start New Session` in client.
```

### Model and Agent

N/A

### Operating System

macOS Sonoma 14.6.1

### Reproduction Steps

1. Clear any previous Docker images and containers related to OpenHands
2. Run `make build && make run`
3. Open client and `Start New Session`
4. Observe terminal

### Logs, Errors, Screenshots, and Additional Context

```sh
Successfully built openhands-0.8.3.tar.gz
12:38:11 - openhands:INFO: runtime_build.py:57 - Source distribution created at /Users/.../Desktop/aha/OpenHands/dist/openhands-0.8.3.tar.gz
12:38:11 - openhands:INFO: runtime_build.py:78 - Source distribution moved to /var/folders/h9/k824x1kx25g4mh3jgf3ld0h00000gn/T/tmpa16zs8l1/project.tar.gz
12:38:11 - openhands:INFO: runtime_build.py:88 - Unpacked source code directory: /var/folders/h9/k824x1kx25g4mh3jgf3ld0h00000gn/T/tmpa16zs8l1/code
12:38:11 - openhands:INFO: runtime_build.py:159 - Input base image: nikolaik/python-nodejs:python3.11-nodejs22
Skip init: False
Extra deps: None
Hash for docker build directory [/var/folders/h9/k824x1kx25g4mh3jgf3ld0h00000gn/T/tmpa16zs8l1] (contents: ['Dockerfile', 'code']): a6d8c497e8c9ea9ad7786861232e1aff

############################################################
# TERMINAL STOPS SENDING DATA FOR THE NEXT ~7 MINUTES FROM THE LINE BELOW
############################################################
12:38:13 - openhands:INFO: docker.py:75 - Cannot pull image ghcr.io/all-hands-ai/runtime:a6d8c497e8c9ea9ad7786861232e1aff directly
############################################################
# DESPITE THE TIME STAMP OF THE FOLLOWING LINE, IT IS ONLY PRINTED TO THE TERMINAL AFTER ~7 MINUTES
############################################################
12:42:13 - openhands:INFO: runtime_build.py:254 - Cannot find docker Image [ghcr.io/all-hands-ai/runtime:a6d8c497e8c9ea9ad7786861232e1aff]
Will try to re-build it from latest [ghcr.io/all-hands-ai/runtime:od_v0.8.3_image_nikolaik___python-nodejs_tag_python3.11-nodejs22] image to potentially save time for dependencies installation.

12:42:13 - openhands:INFO: runtime_build.py:41 - Using project root: /Users/.../Desktop/aha/OpenHands
* Creating isolated environment: venv+pip...
* Installing packages in isolated environment:
  - poetry-core
* Getting build dependencies for sdist...
* Building sdist...
Successfully built openhands-0.8.3.tar.gz
12:42:20 - openhands:INFO: runtime_build.py:57 - Source distribution created at /Users/.../Desktop/aha/OpenHands/dist/openhands-0.8.3.tar.gz
12:42:20 - openhands:INFO: runtime_build.py:78 - Source distribution moved to /var/folders/h9/k824x1kx25g4mh3jgf3ld0h00000gn/T/tmpoed5_ip2/project.tar.gz
12:42:20 - openhands:INFO: runtime_build.py:88 - Unpacked source code directory: /var/folders/h9/k824x1kx25g4mh3jgf3ld0h00000gn/T/tmpoed5_ip2/code
12:42:20 - openhands:INFO: runtime_build.py:159 - Input base image: ghcr.io/all-hands-ai/runtime:od_v0.8.3_image_nikolaik___python-nodejs_tag_python3.11-nodejs22
```

**Comment by tobitege** [comment]
Related: #3440

**Comment by rbren** [comment]
Does this only happen in dev mode? Or does it happen for all users?

**Comment by tobitege** [comment]
It technically happens to all, but is most noticeable, when e.g. your local Docker doesn't have any images yet, so it has to rebuild them all. But even rebuilding one takes more than the "10 seconds" that show up on the UI right now.

**Comment by amanape** [comment]
Technically resolved by #3645 according to the issue title
